### PR TITLE
gnutar: pull missing `libintl` dependency on Darwin

### PR DESCRIPTION
**Pull request by trofi**
Without the change `gnutar` fails the build as:

      "_libintl_setlocale", referenced from:
          _main in tar.o
      "_libintl_textdomain", referenced from:
          _main in tar.o
    ld: symbol(s) not found for architecture x86_64

https://hydra.nixos.org/build/228724258/nixlog/2/tail https://hydra.nixos.org/build/228711328/nixlog/2/tail

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by trofi** [comment]
There is a caveat: on `staging` `darwin` is a bit broken for https://github.com/NixOS/nixpkgs/pull/244400#issuecomment-1646774584 . I tested on `staging-23.0.5`. Builds fine.

File: pkgs/tools/archivers/gnutar/default.nix [hunk 1]
```diff
@@ -31,7 +31,12 @@ stdenv.mkDerivation rec {
   outputs = [ "out" "info" ];
 
   nativeBuildInputs = lib.optional stdenv.isDarwin autoreconfHook;
-  buildInputs = lib.optional stdenv.isLinux acl;
+  # Add libintl on Darwin specifically as it fails to link (or skip)
+  # NLS on it's own:
+  #  "_libintl_textdomain", referenced from:
+  #    _main in tar.o
+  #  ld: symbol(s) not found for architecture x86_64
+  buildInputs = lib.optional stdenv.isLinux acl ++ lib.optional stdenv.isDarwin [ libintl ];
```
Comment (vcunat) [inline pkgs/tools/archivers/gnutar/default.nix:39]: Nit: either use `optionals` or drop the brackets.

Comment (trofi) [reply pkgs/tools/archivers/gnutar/default.nix]: Oh, good point! Dropped brackets.

**Review by vcunat — APPROVED** [review]
Looks good.  And on staging-next-23.05 they now build fine for both `*-darwin` on Hydra.

**Comment by vcunat** [comment]
Maybe it's all non-Linux, but there's little practical difference for us and I don't have a way of trying (which is closely related).

**Pull request merged**